flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/ExecutionFailureHandler.java

```diff
@@ -115,6 +122,29 @@ private FailureHandlingResult handleFailure(
             final Set<ExecutionVertexID> verticesToRestart,
             final boolean globalFailure) {
 
+        if (!globalFailure) {
```

Comment:
I would propose to do the translation of the partition exception of a cached result in `DefaultScheduler` and keep `ExecutionFailureHandler` as is. The benefits are:
1. the `ExecutionFailureHandler` does not need to understand exceptions.
2. no need to modify `DefaultExecutionVertex`. The scheduler can have a method to check if a partition is from a cached result.
3. no need to distinguish if it is a global failure or not. Actually even if it is a global failure, `CachedIntermediateDataSetCorruptedException` should also fail the job in unrecoverable way.

Reply:
You are right. It is much clear to do the translation in the DefaultScheduler. Change is made in [976d73b](https://github.com/apache/flink/pull/19653/commits/976d73b7f33d1d5ee8b788906b083419d6342748).